src/Dialogs/QuickGuide/WelcomeWidget.cpp

```
@@ -0,0 +1,224 @@
+  const PixelRect t4_rc{x, y, rc.right - margin, rc.bottom};
+  const unsigned t4_height = canvas != nullptr
+    ? canvas->DrawFormattedText(t4_rc, t4.c_str(), DT_LEFT)
+    : renderer.GetHeight(look.text_font, text_width, t4.c_str());
+  y += int(t4_height) + margin;
+
+  const TCHAR *t5 = _("https://github.com/XCSoar/XCSoar");
+  if (canvas != nullptr) {
+    canvas->SetTextColor(COLOR_BLUE);
+    canvas->Select(look.text_font);
+  }
+  const PixelRect t5_rc{x, y, rc.right - margin, rc.bottom};
```

Cppcheck (2.19.0)

[error] Dereference of an invalid iterator

(derefInvalidIterator)